### PR TITLE
✨ Add expiration time to experiments

### PR DESCRIPTION
**Pull request by jonathantyng-amp**
Add an expiration time (Unix timestamp in milliseconds) to experiments tracked in `experiments-config.json`. Build will throw an error if expiration timestamp is missing, or if expiration time has passed.

cc @zhouyx for review :)

### CODE REVIEW

File: build-system/global-configs/experiments-config.json [hunk 1]
```diff
@@ -3,6 +3,7 @@
     "name": "ANALYTICS_VENDOR_SPLIT",
     "command": "gulp dist --defineExperimentConstant=ANALYTICS_VENDOR_SPLIT",
     "issue": "",
+    "expirationTimestampMs": "1565870400000",
```
Comment (zhouyx) [inline build-system/global-configs/experiments-config.json]: hmmm the value is not very readable. Is it possible to have the value follow some sort of readable format?

Comment (jonathantyng-amp) [reply build-system/global-configs/experiments-config.json]: we can also do `2019-01-15`, which follows year-month-day format

Comment (zhouyx) [reply build-system/global-configs/experiments-config.json]: Sure I like the idea. May need to also agree on timezone as well. I wonder if there's a UTC date format that we can follow. Could you please take a look? Thanks.

Comment (jonathantyng-amp) [reply build-system/global-configs/experiments-config.json]: Using this date-only format in the `Date` constructor [defaults to UTC timezone](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/parse#Date_Time_String_Format) :)

**Comment by jonathantyng-amp** [comment]
Changed expiration date to YYYY-MM-DD format, in UTC

cc @zhouyx for another review :)

File: build-system/global-configs/README.md [hunk 2]
```diff
@@ -10,5 +10,5 @@ besides 1 or 0.
 - `name`: Experiment name
 - `command`: Command used to build the experiment
 - `issue`: The issue tracker URL for this experiment
-- `expirationTimestampMs`: The Unix timestamp in ms when this experiment expires
+- `expirationDateUTC`: The experiment expiration date in YYYY-MM-DD format, in UTC
```
Comment (zhouyx) [inline build-system/global-configs/README.md]: Thank you for the documentation!

**Comment by zhouyx** [comment]
@jonathantyng-amp Could you please take a look at the travis failure. Thank you

**Comment by jonathantyng-amp** [comment]
@zhouyx ready to merge :)

File: build-system/global-configs/README.md [hunk 3]
```diff
@@ -10,5 +10,5 @@ besides 1 or 0.
 - `name`: Experiment name
 - `command`: Command used to build the experiment
 - `issue`: The issue tracker URL for this experiment
-- `expirationDateUTC`: The experiment expiration date in YYYY-MM-DD format, in UTC
+- `expirationDateUTC`: The experiment expiration date in YYYY-MM-DD format, in UTC. If an experiment is expired, it will fail the build process. There is currently no realtime check for the expiration time.
```
Comment (zhouyx) [inline build-system/global-configs/README.md]: Thanks for clarifying the expiration date. 
Realtime check will be difficult to add and can cause a lot of confusion. I don't believe we will be adding that in the future. Do you think we should rephrase the sentence to not mention the concept? Instead we can explain how the expiration data affects the build process, and note that the experiment will be started and ended the same time along with our release process.

**Comment by zhouyx** [comment]
LGTM. Thank you for adding the expiration time support 🎉

**Pull request merged**